barretenberg/cpp/src/barretenberg/plonk_honk_shared/library/grand_product_library.hpp

```diff
@@ -143,29 +143,21 @@ void compute_grand_product(const size_t circuit_size,
     });
 }
 
+/**
+ * @brief Compute the grand product corresponding to each grand-product relation defined in the Flavor
+ *
+ */
 template <typename Flavor>
-void compute_grand_products(const typename Flavor::ProvingKey& key,
-                            typename Flavor::ProverPolynomials& full_polynomials,
+void compute_grand_products(typename Flavor::ProverPolynomials& full_polynomials,
                             bb::RelationParameters<typename Flavor::FF>& relation_parameters)
 {
     using GrandProductRelations = typename Flavor::GrandProductRelations;
-    using FF = typename Flavor::FF;
 
     constexpr size_t NUM_RELATIONS = std::tuple_size<GrandProductRelations>{};
     bb::constexpr_for<0, NUM_RELATIONS, 1>([&]<size_t i>() {
         using GrandProdRelation = typename std::tuple_element<i, GrandProductRelations>::type;
 
-        // Assign the grand product polynomial to the relevant std::span member of `full_polynomials` (and its shift)
```

Comment:
This confusing nonsense is the epitome of what I wanted to get rid of with this work.